### PR TITLE
feat(waf): enable log waf logging for waf enabled services

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description
Implementation of https://github.com/hamlet-io/engine/pull/1366  which enables log forwarding for WAF enabled services

To setup WAF logging you need 
- A datafeed component deployed with a backend configured ( for s3 no lambda transfromation is required, for Elasticsearch this may be required ) 
  - The datafeed also needs to be configured as a WAFLogFeed using the `aws:WAFLogFeed` configuration attribute. This changes the name of the deployed firehose to align with WAF setup requireemnts 
- A logging profile which links to this datafeed component 
   - Assign this logging profile under the WAF configuration `WAF.Profiles.Logging` 

**note:** if using the same datafeed for multiple WAF endpoints the log files will be stored together, they include a GUID in their file name so this should be safe

if using S3 you can use the following guide from AWS on setting up athena to query the logs
https://docs.aws.amazon.com/athena/latest/ug/waf-logs.html

## Motivation and Context

AWS WAF doesn't provide information out of the box about what caused a given request to fail. Instead they require you to setup a log forwarding solution to analyse the logs with an appropriate tool 

## How Has This Been Tested?
Tested locally across all components

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions
- Core - https://github.com/hamlet-io/engine/pull/1366
- AWS - https://github.com/hamlet-io/engine-plugin-aws/pull/105
- Bash Executor -  https://github.com/hamlet-io/executor-bash/pull/71

## Checklist:
- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

File: aws/services/waf/resource.ftl [hunk 1]
```diff
@@ -311,3 +311,113 @@
...
+                    [/#switch]
+                [/#if]
+            [/#if]
+        [/#list]
+    [/#list]
+
+    [@debug message="WAFLog" context=dataFeedArns enabled=true /]
+
+    [#if dataFeedArns?has_content ]
+        [#if deploymentSubsetRequired("cli", false) ]
```
Comment (ml019) [inline aws/services/waf/resource.ftl:367]: Usually details of subsets are contained in calling setup routines and the service support routines assume they need to do whatever is provided. 

Maybe worth adding the actual subset names as parameters?

Comment (roleyfoley) [reply aws/services/waf/resource.ftl:367]: So are you saying call this macro twice in each component? Once in the setup `cli` subset and once in the setup `epilogue`?

Comment (roleyfoley) [reply aws/services/waf/resource.ftl:367]: So the other approach could be that we have `create` service routines and `setup` routines. Creates are the raw create an output things and `setup` is a more complete functionality which can include multiple subsets

Comment (ml019) [reply aws/services/waf/resource.ftl:367]: The only change I was suggesting was to add the actual subset names as parameters. "cli" and "epilogue" are pretty well wired into the AWS support so I'll merge as is.

**Pull request merged**